### DISCUSSION TITLE
How to use Nuclei in an Internal Environement?

### DISCUSSION THREAD

**Discussion by umar98** [projectdiscovery/nuclei#1995] [Q&A] [answered]
Hello all?

Actually, I'm relatively new to Penetration Testing and in love with Nuclei(I also have started to write my own templates, this project is a love). But my only doubt is, how can I use Nuclei when in an Internal Environment as past won't be available due to no Internet. 

Like I do I create a standalone server for Nuclei so it works in the same way. 

Thank You!

**Comment by ehsandeep** [comment, marked as answer]
@umar98 You can still use the nuclei in the internal network by disabling version update check with `-duc` flag; also you need to disable use of public interactsh server with `-ni` flag and last, you can make use of `-sr` flag to use system resolver to resolve your internal hosts, if that doesn't work, you can also consider using `-r` flag to provider internal dns server to resolve internal hosts.

```console
   -duc, -disable-update-check    disable automatic nuclei/templates update check
   -ni, -no-interactsh            disable interactsh server for OAST testing, exclude OAST based templates
   -sr, -system-resolvers         use system DNS resolving as error fallback
   -r, -resolvers string          file containing resolver list for nuclei
```

I believe this should be sufficient to run nuclei in the internal environment, but let me know if this doesn't work for you.

**Reply by umar98** [reply]
Thanks for the Great tip. Though I have a question if we disable the interactsh server, then as the description states the OAST-based Templates and templates that reply on input to be verified, how will they be executed?

**Reply by ehsandeep** [reply]
Since you mentioned you are looking to use nuclei in an environment where you don't have access to the external internet using public interactsh is also not possible and possibly you can set up [self-hosted](https://github.com/projectdiscovery/interactsh#interactsh-server) interactsh server which is accessible in your network and can be used using `-iserver` flag.

**Reply by umar98** [reply]
Exactly what I was looking for. Thank You. 

(I might also be pinging you again if I need any help. xD)

Once again, Thanks!

**Reply by ehsandeep** [reply]
Feel free to join the discord server - https://discord.gg/projectdiscovery
